src/library/dlgtrackinfo.cpp

```diff
@@ -402,7 +428,9 @@ void DlgTrackInfo::saveTrack() {
     m_pLoadedTrack->setTrackNumber(txtTrackNumber->text());
     m_pLoadedTrack->setComment(txtComment->toPlainText());
 
-    if (!m_pLoadedTrack->isBpmLocked()) {
+    m_pLoadedTrack->setBpmLocked(beatgridLock->checkState() == Qt::Checked);
+
+    if (m_beatsChanged) {
         m_pLoadedTrack->setBeats(m_pBeatsClone);
         reloadTrackBeats(*m_pLoadedTrack);
```

Comment:
Please reset m_beatsChange here instead of in an unrelated context when leaving this function.